src/it/scala/com/wavesplatform/it/Node.scala

```diff
@@ -0,0 +1,155 @@
+
+  def post(path: String, f: RequestBuilder => RequestBuilder = identity): Future[Response] =
+    retrying(f(
+      _post(s"http://localhost:${nodeInfo.hostRestApiPort}$path").setHeader("api_key", "integration-test-rest-api")
+    ).build())
+
+  def post[A: Writes](path: String, body: A): Future[Response] =
+    post(path, (rb: RequestBuilder) => rb.setHeader("Content-type", "application/json").setBody(stringify(toJson(body))))
+
+  def connectedPeers: Future[Seq[Peer]] = get("/peers/connected").map { r =>
+    (Json.parse(r.getResponseBody) \ "peers").as[Seq[Peer]]
+  }
+
+  def height: Future[Long] = get("/blocks/height").as[JsValue].map(v => (v \ "height").as[Long])
+  def blockAt(height: Long) = get(s"/blocks/at/$height").as[Block]
+  def lastBlock: Future[Block] = get("/blocks/last").as[Block]
+  def blockSeq(from: Long, to: Long) = get(s"/blocks/seq/$from/$to").as[Seq[Block]]
+  def status: Future[Status] = get("/node/status").as[Status]
+  def balance(address: String): Future[Balance] = get(s"/addresses/balance/$address").as[Balance]
+  def transfer(sourceAddress: String, recipient: String, amount: Long, fee: Long): Future[Transaction] =
+    post("/assets/transfer", TransferRequest(None, None, amount, fee, sourceAddress, None, recipient)).as[Transaction]
+
+  def createAlias(targetAddress: String, alias: String, fee: Long): Future[Transaction] =
+    post("/alias/create", CreateAliasRequest(targetAddress, alias, fee)).as[Transaction]
+
+  def waitForHeight(targetHeight: Long): Future[Long] =
+    timer.retryUntil[Long](height, _ >= targetHeight, generationDelay)
+  def waitForNextBlock: Future[Block] = for {
+    currentBlock <- lastBlock
+    actualBlock <- findBlock(_.height > currentBlock.height, currentBlock.height)
+  } yield actualBlock
+
+  def findBlock(cond: Block => Boolean, from: Long = 1, to: Long = Long.MaxValue): Future[Block] = {
+    val p = Promise[Block]
+
+    def reschedule(_from: Long, _to: Long) = try {
+      timer.newTimeout(retrying(_from, _to), generationDelay.toMillis, MILLISECONDS)
+    } catch {
+      case NonFatal(t) => p.failure(t)
+    }
+
+    def retrying(_from: Long, _to: Long)(timeout: Timeout): Unit = blockSeq(_from, _to).onComplete {
+      case Success(blocks) =>
+        blocks.find(cond) match {
+          case Some(b) =>
+            log.debug(s"Found matching block ${b.signature}")
+            p.success(b)
+          case None =>
+            val newFrom = blocks.lastOption.fold(_from)(_.height + 1)
+            if (newFrom > to) {
+              p.failure(new NoSuchElementException)
+            } else {
+              log.debug(s"Loaded ${blocks.length} blocks, no match found. Next range: [$newFrom, ${newFrom + 99}]")
+              reschedule(newFrom, (newFrom + 99).min(to))
+            }
+        }
+      case Failure(t) =>
+        log.debug("Error loading blocks", t)
+        reschedule(_from, _to)
+    }
+
+    retrying(from, from + 99)(null)
```

Comment:
It's a trick to use a function where an instance implementing a functional [TimerTask](https://netty.io/4.0/api/io/netty/util/TimerTask.html) interface is required. In this very case, null is like a seed for a (semi-)recursive function.